src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/CollectionSortClauseBuilder.java

```diff
@@ -21,8 +26,10 @@ public CollectionSortClauseBuilder(CollectionSchemaObject collection) {
   }
 
   @Override
-  public SortClause buildAndValidate(ObjectNode sortNode) {
-    // $lexical is only special for Collections: handle first
+  public SortClause buildClauseFromDefinition(ObjectNode sortNode) {
```

Comment:
nit: consider a short circuit to exit the object node is empty, I am guessing that would happen a lot for findOne

Reply:
I can add that, sure.